### DISCUSSION TITLE
I want to change the content of H1 to make it more user friendly. What will Google see?

### DISCUSSION THREAD

**Discussion by Vinc26** [GetPublii/Publii#1735] [Help]
Hi,

The H1 tag on my home page must be perfect for Google. But it will not be very suitable for the user: for him, I want a slightly more poetic text. So I add an html block in my page:

`<script>document.getElementsByTagName("h1")[0].textContent = "My new text here !";</script>`

It works. But how can I be sure that Google will see my original text? What do you think?

**Comment by bjazmoore** [comment]
The H1 is the page title.  What are you doing different here than renaming the page dynamically when viewed?  Is there a problem just using the text you are inserting as the page/post title?

**Reply by Vinc26** [reply]
The H1 is the page title. In the case of my home page, it is my name and my job. But the visitor know who I am and what I'm offering. So, I prefer dynamically change the H1 text by something more interesting for the visitor. But I want Google keep only my name and job.
It's like Apple website. The H1 is hidden. But it has to be here for SEO.
`<h1 class="visuallyhidden">Apple</h1>`

**Reply by bjazmoore** [reply]
You probably should invest in [smart replacer](https://marketplace.getpublii.com/plugins/smart-replacer/).

**Reply by riker09** [reply]
My days as SEO practicioner are long over, but I do remember that tricking Google almost never is a good idea. Google _will_ evaluate your JS, even when they say they won't. At best everything works as intended, at worst your website gets flagged for using this behaviour in some way.

If it was my website I would spent some time to make the content suitable for both use cases.

**Comment by candidexmedia** [comment]
@Vinc26 I think you might be confusing your H1 page title with the page's HTML `<title>` tag: 
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/title
- https://www.w3schools.com/tags/tag_title.asp
- https://moz.com/learn/seo/title-tag

![image](https://github.com/user-attachments/assets/c129d0cf-e49b-402d-bf54-a9cfb7f2268e)


The title you set on the post editor is the title that's visible onscreen when your visitor accesses that page, but isn't the title used in search engine results or browser tabs.

The H1 page/post title is the one is set at the very top of every post/page you make:
![image](https://github.com/user-attachments/assets/3d20f150-30a5-468f-968b-e2d795af3daa)


The `<title>` is set in your Post Settings (right sidebar in the post editor, under the "Page title" field). That one is displayed in the browser tab, and on search result pages (along with the metadescription), and *not* on your page:

![image](https://github.com/user-attachments/assets/d8bbb691-c22b-4938-ae2d-f5303c80ae5b)

**Reply by bjazmoore** [reply]
Great explanation.  Thanks

**Comment by Vinc26** [comment]
Thank you for your explanations and links. I already know the differences between H1 and TITLE. But it was an opportunity for me to deepen my knowledge of SEO. Thanks :). I have read other articles. and unfortunately, no, Google does not systematically use TITLE for its search results; if it considers TITLE not good enough, it will use H1, and eventually rewrite one of them... And it's not easy to make a TITLE that Google likes.
Here is a very interesting article (and its comments too): [https://dev.to/johndotowl/why-your-h1-and-title-should-be-different-for-seo-349m](https://dev.to/johndotowl/why-your-h1-and-title-should-be-different-for-seo-349m)
Apart from that, yes, in Publii, whether it is a page or a post, H1 is the title of the content at the top, while TITLE is defined in the page settings and will be the name of the window or tab.

That's why I want to take care of my H1 in case it is used by Google for the result. And therefore change its display at the last moment on the user's browser, so that it is shorter and more user friendly. The question is therefore always the same: will my javascript rewriting trick be ignored by Google?

**Reply by candidexmedia** [reply]
Thank you for sharing that blog! I didn't know that Google could potentially ignore the title tag, because it usually picks it up for my pages.

I would suggest consulting Google's docs directly for directives: https://developers.google.com/search/docs/appearance/title-link#page-titles

H1 is only mentioned three times, and one of those times talks about headings in general, among a list of other elements that *could* be considered. It sounds like a well formatted `<title>` tag holds more weight (if I'm reading correctly).

> The question is therefore always the same: will my javascript rewriting trick be ignored by Google?

This forum might not be the right avenue to get insight on this. A dedicated SEO forum, web dev forum, or Google forum will probably provide better info. I'll admit: I've never come across that kind of script for that purpose, but someone elsewhere might have.

It might also be worth setting up a test site and simply experimenting to see what Google picks up.
